### DISCUSSION TITLE
:prImmediately seems not work well (?)

### DISCUSSION THREAD

**Discussion by exKAZUu** [renovatebot/renovate#9724] [Request Help] [answered]
Hi all,

I maintain a OSS repository: https://github.com/WillBooster/plantuml-visualizer
This repo refers to our configuration file via devDependencies (https://github.com/WillBooster/plantuml-visualizer/blob/main/package.json#L57)
The configuration file includes `:prImmediately` (https://github.com/WillBooster/willbooster-configs/blob/665c3c04a83d958bb4ddbdf3a75deba5b5aba839/packages/renovate-config/package.json#L26), but the branches created by renovate are not shown as PRs.

- PRs: https://github.com/WillBooster/plantuml-visualizer/pulls
- `renovate/all-minor-patch`: https://github.com/WillBooster/plantuml-visualizer/tree/renovate/all-minor-patch
- `renovate/lock-file-maintenance`: https://github.com/WillBooster/plantuml-visualizer/tree/renovate/lock-file-maintenance

How can we ask renovate to create PRs immediately after creating branches?

---

The answer should be refferred is https://github.com/renovatebot/renovate/discussions/9724#discussioncomment-717930

**Comment by rarkins** [comment, marked as answer]
What do the logs say for these branches? Log in at https://app.renovatebot.com

FYI `:prImmediately` actually represents the *default* behavior so it's not even needed - PRs should default to being created immediately.

**Reply by exKAZUu** [reply]
@rarkins  Thank you for the quick reply. Here are two logs (12 hours and 13 hours ago). It is now 15:12 (Mon) at Tokyo.
The scheduled time is `3am on Monday, Asia/Tokyo`, so I belive renovate should work on either job, but both jobs seem not work.

---

Around 02:00 AM on Monday in UTC+0900

[log-13hours-ago.log](https://github.com/renovatebot/renovate/files/6374652/log-13hours-ago.log)

```
DEBUG: Checking schedule "before 3am on Monday"(branch="renovate/all-minor-patch")
{
  "parsedSchedule": {
    "schedules": [
      {
        "t_b": [
          10800
        ],
        "d": [
          2
        ]
      }
    ],
    "exceptions": [],
    "error": -1
  }
}
DEBUG: Does not match schedule because 7 is not in 1(branch="renovate/all-minor-patch")
```

---

After 03:00 AM on Monday in UTC+0900

[log-12hours-ago.log](https://github.com/renovatebot/renovate/files/6374651/log-12hours-ago.log)

```
DEBUG: Checking schedule "before 3am on Monday"(branch="renovate/all-minor-patch")
{
  "parsedSchedule": {
    "schedules": [
      {
        "t_b": [
          10800
        ],
        "d": [
          2
        ]
      }
    ],
    "exceptions": [],
    "error": -1
  }
}
DEBUG: Does not match schedule because 10916 is later than 10800(branch="renovate/all-minor-patch")
```

**Reply by exKAZUu** [reply]
@rarkins I've changed the scheduled time from `schedule:weekly` to `"schedule": [ "before 10am on Saturday" ]`, then renovate works well. So, I suspect that renovate cannot be launched if `X` < `Y` and `"before X am on Monday"` is applied in the area of `GMT+Y`. (e.g. `before 3am on Monday` in Tokyo GTM+9)

**Reply by viceice** [reply]
Can you post the exact times when renovate did run? Renovate is running hourly but not on an exact schedule. Do maybe your first run already passed 3am on Monday.

Maybe we have a timezone issue in later?

**Reply by viceice** [reply]
I also found on logs that the first run logged 1800 on Sunday in UTC+0000. So it already passed 0300 on Monday in UTC+0900.

**Reply by exKAZUu** [reply]
@viceice I'm not sure how I can check the times when renovate run, but, `log-12hours-ago.log` includes `10916 is later than 10800`, so I expect it was after 03:00 AM on Monday in UTC+0900. Do you need more information?

**Reply by viceice** [reply]
So when renovate has run shortly after 3 am then it's expected that the schedule fails for your 12h ago job. but it the job at 2am should match the schedule. 

So it seems again like a timezone issue when checking schedules.
It looks like we get the right time but a wrong day.

**Reply by viceice** [reply]
I've made a small pr to add some more tests, but can't reproduce https://github.com/renovatebot/renovate/pull/9860

@exKAZUu Can you past new logs from this monday from the relevant times?

**Reply by exKAZUu** [reply]
@viceice Sorry, I now cannot see the logs of the last Monday. I will upload the next Monday's log here.

**Reply by exKAZUu** [reply]
@viceice hmm, I've changed our shared config (https://github.com/WillBooster/willbooster-configs/commit/441eafedf7e6cb9472c54b39331cf4fd6fbe39a0). Though many our OSS repos haven't refer to the new version of the shared config yet, renovate seems work in some our repos at 10 am on Saturday (in Tokyo). I'm confused now.

e.g. the following repos have renovate PRs created on Saturday
- https://github.com/WillBooster/plantuml-visualizer/pulls
- https://github.com/WillBooster/gen-i18n-ts/pulls

But, https://github.com/WillBooster/markdown-it-img-size/pulls seems still not work. The repo refers to https://github.com/WillBooster/willbooster-configs/blob/0ba28b9c9143e76813c6b490660edf69a12978c0/packages/renovate-config/package.json .
Some renovate branches (https://github.com/WillBooster/markdown-it-img-size/branches, e.g. `renovate/all-minor-patch`) in the repo are still not converted into PRs.
Only one log record exists in https://app.renovatebot.com as follows

[renovate-10th-May-Around-6am-UTC+9.log](https://github.com/renovatebot/renovate/files/6449007/renovate-10th-May-Around-6am-UTC%2B9.log)

**Reply by viceice** [reply]
I think your problem was that the last commit was from 21.02.2021. Renovate will reduce to weekly runs when it detects no changes to default branch. So it won't run at your schedule. I've seen you merged a pr now, so renovate should now run hourly again.

@rarkins Can we document this app behavior somewhere in the docs?

**Reply by exKAZUu** [reply]
Okay, I feel renovate are working well now. Thank you for the support!
